rest-api.md

```diff
@@ -0,0 +1,263 @@
+
+## Configuration
+
+Vous pouvez continuer à travailler dans le même projet `prisma-workshop` que vous avez mis en place dans la leçon 1. Cependant, le starter de cette leçon est situé dans la branche `rest-api` du repo que vous avez cloné.
+
+Avant de passer à cette branche, vous devez commiter l'état actuel de votre projet. Pour simplifier, vous pouvez utiliser la commande `git stash` pour le faire :
+
+```bash
+git stash
+```
+
+Après avoir exécuté cette commande, vous pouvez passer à la branche `rest-api` et supprimer le répertoire `migrations` et le fichier `dev.db` :
+
+```bash
+git checkout rest-api
+rm -rf prisma/migrations
+rm prisma/dev.db
+```
+
+Ensuite, enlevez vos dépendances npm et réinstallez-les pour tenir compte des nouvelles dépendances dans le `package.json` :
+
+```bash
+rm -rf node_modules
+npm install
+```
+
+Le modèle de données que vous allez utiliser ici est très similaire à celui que vous avez créé précédemment, mais le modèle `Post` a été étendu avec quelques champs supplémentaires :
+
+```graphql
+model User {
+  id    Int     @id @default(autoincrement())
+  email String  @unique
+  name  String?
+  posts Post[]
+}
+
+model Post {
+  id        Int      @id @default(autoincrement())
+  **createdAt DateTime @default(now())
+  updatedAt DateTime @updatedAt**
+  title     String
+  content   String?
+  published Boolean  @default(false)
+  **viewCount Int      @default(0)**
+  author    User?    @relation(fields: [authorId], references: [id])
+  authorId  Int?
+}
+```
+
+Puisque vous recommencez votre installation de Prisma, vous devez recréer la base de données et ses tables. Exécutez la commande suivante :
+
+```bash
+npx prisma migrate dev --name init
+```
+
+Enfin, vous pouvez amorcer la base de données avec un échantillon de données qui est spécifié dans le fichier `prisma/seed.ts`. Vous pouvez exécuter ce script d'amorçage avec cette commande :
+
+```bash
+npx prisma db seed --preview-feature
+```
+
+Voilà, vous êtes prêt pour vos tâches maintenant !
+
+## Tâches
+
+Vous pouvez trouver les tâches pour cette leçon à l'intérieur du fichier `src/index.ts` marqué avec `TODO`. Votre objectif est d'insérer les bonnes requêtes Prisma Client pour chaque route de l'API REST.
+
+Notez que ceci n'est *pas* une leçon sur la conception d'API et que vous devriez réfléchir plus attentivement à la conception de vos opérations d'API dans une application.
+
+Si vous utilisez VS Code, vous pouvez installer l'extension [REST Client](https://marketplace.visualstudio.com/items?itemName=humao.rest-client) et tester votre implémentation en utilisant les appels HTTP fournis dans [`test.http`](https://github.com/nikolasburk/prisma-workshop/blob/rest-api/test.http).
+
+- Voir une démo rapide de l'extension VS Code REST Client
+    
+    ![3%20REST%20API%204973ea32555049049a2a5e24a026e768/rest-client.gif](3%20REST%20API%204973ea32555049049a2a5e24a026e768/rest-client.gif)
```

Comment:
Same for here, I think we need to manually upload the images/gifs in this repo and link to it

Reply:
I have uploaded and linked to the image